### PR TITLE
Stop using mx_GroupLayout for styling

### PR DESCRIPTION
**Pull request by luixxiul**
For https://github.com/vector-im/element-web/issues/22434

For `mx_EventTile`, `mx_GroupLayout` is globally a parent element. Because `mx_GroupLayout` includes `mx_EventTile` as a child element on _GroupLayout.scss, any rules to style `mx_EventTile` differently require to have `mx_GroupLayout` as a parent selector in order to cancel cascading effect (see: [ThreadsList](https://github.com/matrix-org/matrix-react-sdk/blob/2649c6fd7557cde13f0d852fcc9696effb19435a/res/css/views/right_panel/_ThreadPanel.scss#L96)), which makes style rules unnecessarily complicated and bloated.

Since `mx_GroupLayout` has currently `mx_EventTile` as a sole direct child element on _GroupLayout.scss, it should be able to be safely removed in favor of `mx_EventTile[data-layout=group]`, which is used much more widely to style EventTile on the group layout. With this PR, `mx_GroupLayout` is used *only one time* on [_EventTilePreview.scss](https://github.com/matrix-org/matrix-react-sdk/blob/bae8854c13b2ae7b3452a6c79b70debb87433fa2/res/css/views/elements/_EventTilePreview.scss#L19) :tada:

Actually _GroupLayout.scss can be (and should be) deleted altogether by moving the rest of the rules on the file because there is no "Group Layout", but it is out of scope of this PR.

type: task

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by luixxiul** [comment]
Though the change itself is fairly small, I've split it into meaningful chunks for the same of reviewers. Please ask me if there is something which is not clear enough. I could provide some screenshots of DOM if requested.

File: res/css/views/rooms/_EventTile.scss [hunk 1]
```diff
@@ -128,7 +145,6 @@ $threadInfoLineHeight: calc(2 * $font-12px); // See: _commons.scss
         display: inline-block;
         padding-bottom: 0px;
         padding-top: 0px;
-        margin: 0px;
```
Comment (luixxiul) [inline res/css/views/rooms/_EventTile.scss:131]: - For modern layout: specified with https://github.com/matrix-org/matrix-react-sdk/pull/8711/files#diff-bd83194d78de871bc4ad1c4e8d43102c4368481bbb90705152d3d17fea813919R71
- For IRC layout: specified with https://github.com/matrix-org/matrix-react-sdk/pull/8711/files#diff-29c6a3cc88dee2f8be7fa065c46d7d64f66ba006cd8ba9f3f16f301f41d67444R176
- For bubble layout: inline start margin is not required

File: res/css/views/rooms/_EventTile.scss [hunk 2]
```diff
@@ -803,7 +819,7 @@ $threadInfoLineHeight: calc(2 * $font-12px); // See: _commons.scss
     }
 
     .mx_DisambiguatedProfile {
-        margin-right: $spacing-12;
+        margin-inline: 0 $spacing-12;
```
Comment (luixxiul) [inline res/css/views/rooms/_EventTile.scss:822]: Based on https://github.com/matrix-org/matrix-react-sdk/pull/8711/files#diff-f6d1464858812e6ea15c65a49935a089650e5e7adc0a380eeabfce5da896aa67L130

**Review by SimonBrandner — APPROVED** [review]
Looks sane

**Comment by luixxiul** [comment]
mm `cypress: 12-spotlight/spotlight.spec.ts` keeps failing...

**Pull request merged**